packages/kit/src/core/adapt/builder.js

```diff
@@ -0,0 +1,171 @@
+import { SVELTE_KIT } from '../constants.js';
+import { copy, rimraf, mkdirp } from '../../utils/filesystem.js';
+import { prerender } from './prerender.js';
+import { generate_manifest } from '../generate_manifest/index.js';
+
+/**
+ * @param {{
+ *   cwd: string;
+ *   config: import('types/config').ValidatedConfig;
+ *   build_data: import('types/internal').BuildData;
+ *   log: import('types/internal').Logger;
+ * }} opts
+ * @returns {import('types/config').Builder}
+ */
+export function create_builder({ cwd, config, build_data, log }) {
+	/** @type {Set<string>} */
+	const prerendered_paths = new Set();
+	let generated_manifest = false;
+
+	/** @param {import('types/internal').RouteData} route */
+	function not_prerendered(route) {
+		if (route.type === 'page' && route.path) {
+			return !prerendered_paths.has(route.path);
+		}
+
+		return true;
+	}
+
+	return {
+		log,
+		rimraf,
+		mkdirp,
+		copy,
+
+		createEntries(fn) {
+			generated_manifest = true;
+
+			const { routes } = build_data.manifest_data;
+
+			/** @type {import('types/config').RouteDefinition[]} */
+			const facades = routes.map((route) => ({
+				type: route.type,
+				segments: route.segments,
+				pattern: route.pattern,
+				methods: route.type === 'page' ? ['get'] : build_data.server.methods[route.file]
+			}));
+
+			const seen = new Set();
+
+			for (let i = 0; i < routes.length; i += 1) {
+				const route = routes[i];
+				const { id, filter, complete } = fn(facades[i]);
+
+				if (seen.has(id)) continue;
+				seen.add(id);
+
+				const group = [route];
+
+				// figure out which _lower priority_ routes should be considered fallbacks
```

Comment:
does VS Code render markdown in code comments? it makes it harder to read on Github at least
```suggestion
				// figure out which lower priority routes should be considered fallbacks
```